File: Strings/49.Group_Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string , vector<string>> m;
        for(string &s : strs)
        {
            string s1= s;
            sort(s1.begin(),s1.end());
            m[s1].push_back(s);
        }
        vector<vector<string>> ans;
        for(auto i:m)
        {
            ans.push_back(i.second);
        }
        return ans;
    }
};

File: Strings/49. Group_Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        w={}
        l=[]

        for i in strs:
            c=''.join(sorted(i))
            if c not in w:
                w[c]=[]
            w[c].append(i)
        
        for j in w:
            l.append(w[j])

        return l
